### PR TITLE
Restructure folders, add copilot chat to the main solution file

### PR DESCRIPTION
**Pull request by craigomatic**
### Motivation and Context
Clean up the folder structure.

### Description
Aligns with the existing folder structure and adds the CosmosDB memory store implementation to the SKWebApi

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/Directory.Packages.props [hunk 1]
```diff
@@ -4,39 +4,33 @@
...
     <PackageVersion Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
-
     <!-- Microsoft.Extensions.Configuration -->
     <PackageVersion Include="Microsoft.Extensions.Configuration" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Configuration.Binder" Version="7.0.4" />
     <PackageVersion Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Configuration.Json" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Configuration.UserSecrets" Version="7.0.0" />
-
```
Comment (dluc) [inline dotnet/Directory.Packages.props:23]: pls undo these changes

**Comment by dluc** [comment]
could you check the some build warnings? see `dotnet -c Release`

File: dotnet/Directory.Packages.props (hunk 1 above)
Comment (craigomatic) [reply dotnet/Directory.Packages.props]: Whitespace should be returned now

**Comment by craigomatic** [comment]
> could you check the build warnings? see `dotnet -c Release`

I cleaned up one of them, the others we will clean up when we create the PR to main. 

The goal with this PR is to enable the end-to-end bug bash via an aligned folder structure for the samples and some inline comments on how to wire up CosmosDB

**Review by hathind-ms — APPROVED** [review]
